WooCommerce/WooCommerceTests/ViewRelated/Dashboard/DashboardViewModelTests.swift

```diff
@@ -658,6 +658,82 @@ final class DashboardViewModelTests: XCTestCase {
         XCTAssertFalse(topPerformersCard.enabled)
     }
+
+    // MARK: Show New Cards Notice
+
+    func test_showNewCardsNotice_is_false_when_all_new_cards_are_already_in_saved_cards() async {
+        // Given
+        let viewModel = DashboardViewModel(siteID: sampleSiteID, stores: stores)
+        let completeCardsSet: [DashboardCard] = [
+            .init(type: .inbox, availability: .show, enabled: true),
+            .init(type: .reviews, availability: .show, enabled: true),
+            .init(type: .coupons, availability: .show, enabled: true),
+            .init(type: .stock, availability: .show, enabled: true),
+            .init(type: .lastOrders, availability: .show, enabled: true)
+        ]
+        mockLoadDashboardCards(withStoredCards: completeCardsSet)
+
+        // When
+        viewModel.refreshDashboardCards()
+
+        // Then
+        waitUntil {
+            viewModel.dashboardCards.isNotEmpty
+        }
+        XCTAssertFalse(viewModel.showNewCardsNotice)
+    }
+
+    func test_showNewCardsNotice_is_true_when_not_all_new_cards_are_in_saved_cards() async {
+        // Given
+        let viewModel = DashboardViewModel(siteID: sampleSiteID, stores: stores)
+        let incompleteNewCardsSet: [DashboardCard] = []
+        mockLoadDashboardCards(withStoredCards: incompleteNewCardsSet)
+
+        // When
+        viewModel.refreshDashboardCards()
+
+        // Then
+        waitUntil {
+            viewModel.dashboardCards.isNotEmpty
+        }
+
+        XCTAssertTrue(viewModel.showNewCardsNotice)
+    }
+
+    func test_showNewCardsNotice_changes_from_true_to_false_after_showing_customize_screen_and_dismissing_it() async {
+        // Given
+        let incompleteNewCardsSet: [DashboardCard] = [
+            .init(type: .inbox, availability: .show, enabled: false),
+            .init(type: .reviews, availability: .show, enabled: false)
+        ]
+        let viewModel = DashboardViewModel(siteID: sampleSiteID, stores: stores)
+        mockLoadDashboardCards(withStoredCards: incompleteNewCardsSet)
+
+        // When
+        viewModel.refreshDashboardCards()
+
+        waitUntil {
+            viewModel.dashboardCards.isNotEmpty
+        }
+
+        XCTAssertTrue(viewModel.showNewCardsNotice)
+        viewModel.showCustomizationScreen() // Simulate showing Customize screen
+
+        // Simulate saving complete cards since initially we mocked it with empty array
+        let completeCardsSet: [DashboardCard] = [
+            .init(type: .inbox, availability: .show, enabled: true),
+            .init(type: .reviews, availability: .show, enabled: true),
+            .init(type: .coupons, availability: .show, enabled: true),
+            .init(type: .stock, availability: .show, enabled: true),
+            .init(type: .lastOrders, availability: .show, enabled: true)
+        ]
+        mockLoadDashboardCards(withStoredCards: completeCardsSet)
```

Comment:
IMO this test is redundant because you injected the complete set to the load card request, making this test always pass. It doesn't test the functionality of `showCustomizationScreen`. Instead, you should check that the `setDashboardCards` method is triggered with a complete set of cards. Does that make sense?

Reply:
So the reason I had to inject the `completeCardsSet` is because the previous call of `        mockLoadDashboardCards(withStoredCards: incompleteNewCardsSet)` makes `AppSettingsAction.loadDashboardCards` always return incomplete card.

In the test, when  `viewModel.handleCustomizationDismissal()` is called which then does `viewModel.configureNewCardsNotice()`, it will call `loadDashboardCards()` again. In reality this will load the complete card set, but in this test it will return incomplete due to the first mock above. Thus, to mimic reality, I had to do the second inject manually first.

Reply:
I'll merge this for now and if you have a further comment I'll take care of it in a new PR 👍🏼